homeassistant/components/frontier_silicon/browse_media.py

```diff
@@ -0,0 +1,242 @@
+"""Support for media browsing."""
+import asyncio
+import logging
+
+from homeassistant.components.media_player import BrowseError, BrowseMedia
+from homeassistant.components.media_player.const import (
+    MEDIA_CLASS_CHANNEL,
+    MEDIA_CLASS_DIRECTORY,
+    MEDIA_TYPE_CHANNEL,
+)
+
+CHILD_TYPE_MEDIA_CLASS = {
+    MEDIA_TYPE_CHANNEL: MEDIA_CLASS_CHANNEL,
+    "preset": MEDIA_CLASS_CHANNEL,
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class UnknownMediaType(BrowseError):
+    """Unknown media type."""
+
+
+async def build_item_response(media_library, payload, get_thumbnail_url=None):
+    """Create response payload for the provided media query."""
+    search_id = payload["search_id"]
+    search_type = payload["search_type"]
```

Comment:
Why is payload a dictionary if you only use 2 fields? Let's convert that to params.